### PR TITLE
Improve service

### PR DESCRIPTION
**Pull request by secret2830**
1. Add RestrictedServiceFeeDenom param which indicates if the service fee only accepts the base denom
2. Change slashed fees flow from burned to transferred to the fee collector
3. [Breaking change]Update event type `response_service` to `respond_service`

### CODE REVIEW

File: modules/service/keeper/binding.go [hunk 1]
```diff
@@ -123,7 +112,7 @@ func (k Keeper) SetServiceBindingForGenesis(
 	k.SetOwner(ctx, provider, owner)
 	k.SetOwnerProvider(ctx, owner, provider)
 
-	pricing, err := k.ParsePricing(ctx, svcBinding.Pricing)
+	pricing, err := types.ParsePricing(svcBinding.Pricing)
```
Comment (dreamer-zq) [inline modules/service/keeper/binding.go:115]: `types.ParsePricing(svcBinding.Pricing)` is best placed before` k.SetServiceBinding(ctx, svcBinding)`, which can reduce gas

File: modules/service/types/binding.go [hunk 2]
```diff
@@ -72,8 +73,27 @@ func GetDiscountByVolume(pricing Pricing, volume uint64) sdk.Dec {
...
+	if err := json.Unmarshal([]byte(pricing), &rawPricing); err != nil {
+		return p, sdkerrors.Wrapf(ErrInvalidPricing, "failed to unmarshal the pricing: %s", err.Error())
+	}
+
+	priceCoin, err := sdk.ParseCoinNormalized(rawPricing.Price)
+	if err != nil {
+		return p, sdkerrors.Wrapf(ErrInvalidPricing, "invalid price: %s", err.Error())
+	}
+
+	p.Price = sdk.Coins{priceCoin}
```
Comment (dreamer-zq) [inline modules/service/types/binding.go:88]: shoule be use `sdk.NewCoins()`

Comment (secret2830) [reply modules/service/types/binding.go:88]: Zero coins allowed here.

**Review by dgsbl — APPROVED** [review]

**Review by chengwenxi — APPROVED** [review]

**Comment by chengwenxi** [comment]
Could you fix the conflicts? @secret2830

File: modules/service/spec/04_params.md [hunk 3]
```diff
@@ -6,15 +6,15 @@ order: 4
...
-| ComplaintRetrospect  | string (time ns) | "172800000000000"                    |
-| ArbitrationTimeLimit | string (time ns) | "172800000000000"                    |
-| TxSizeLimit          | uint64           | 4000                                 |
-| BaseDenom            | string           | "stake"                              |
-| RestrictedServiceFeeDenom            | bool           | false                              |
+| Key                       | Type             | Example                                       |
+| :------------------------ | :--------------- | :-------------------------------------------- |
+| MaxRequestTimeout         | int64            | 100                                           |
+| MinDepositMultiple        | int64            | 1000                                          |
+| MinDeposit                | array (coins)    | \[{"denom": "uiris","amount": "5000000000"}\] |
```
Comment (chengwenxi) [inline modules/service/spec/04_params.md:13]: why need the `\`?

**Review by chengwenxi — APPROVED** [review]

**Pull request merged**